api/server/router/network/network.go

```diff
@@ -1,26 +1,50 @@
 package network
 
 import (
-	"github.com/docker/docker/api/server/httputils"
 	"github.com/docker/docker/api/server/router"
+	"github.com/docker/docker/api/server/router/local"
+	"github.com/docker/docker/daemon"
+	"github.com/docker/libnetwork"
 )
 
 // networkRouter is a router to talk with the network controller
 type networkRouter struct {
-	routes []router.Route
+	daemon        *daemon.Daemon
+	netController libnetwork.NetworkController
+	routes        []router.Route
 }
 
-// Routes returns the available routes to the network controller
-func (n networkRouter) Routes() []router.Route {
-	return n.routes
+// NewRouter initializes a new network router
+func NewRouter(d *daemon.Daemon) router.Router {
+	n := d.NetworkController()
+	if n == nil {
+		// Return an empty router in cases were controlling networks is not supported, like windows.
+		return &networkRouter{}
+	}
+
+	r := &networkRouter{
+		daemon:        d,
+		netController: n,
+	}
+	r.initRoutes()
+	return r
 }
 
-type networkRoute struct {
-	path    string
-	handler httputils.APIFunc
+// Routes returns the available routes to the network controller
+func (r *networkRouter) Routes() []router.Route {
+	return r.routes
 }
 
-// Handler returns the APIFunc to let the server wrap it in middlewares
-func (l networkRoute) Handler() httputils.APIFunc {
-	return l.handler
+func (r *networkRouter) initRoutes() {
+	r.routes = []router.Route{
+		// GET
+		local.NewGetRoute("/networks", r.getNetworksList),
+		local.NewGetRoute("/networks/{id:.*}", r.getNetwork),
+		// POST
+		local.NewPostRoute("/networks/create", r.postNetworkCreate),
+		local.NewPostRoute("/networks/{id:.*}/connect", r.postNetworkConnect),
```

Comment:
its highly likely that we will expand the connect data to more than container-id (such as alias and other endpoint specific data). When it comes, we dont want to change the API again. So, I prefer to keep the current structure and it can easily be expanded by having backward compatible clients.